server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueMetricContract.java

```diff
@@ -0,0 +1,178 @@
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.api;
+
+import static org.apache.james.queue.api.Mails.defaultMail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.stream.IntStream;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.metrics.api.Gauge;
+import org.apache.james.metrics.api.GaugeRegistry;
+import org.apache.james.metrics.api.Metric;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import com.github.fge.lambdas.Throwing;
+
+public interface MailQueueMetricContract extends MailQueueContract {
+
+    GaugeRegistry gaugeRegistry();
+    Metric enqueuedMailsMetric();
+    Metric dequeuedMailsMetric();
+
+    default FakeMail fakeMail() throws MessagingException {
+        return defaultMail()
+            .name("name1")
+            .build();
+    }
+
+    default void enQueueMail(Integer times) throws Exception {
+        MailQueue mailQueue = getMailQueue();
+
+        IntStream
+            .rangeClosed(1, times)
+            .forEach(Throwing.intConsumer(time -> mailQueue.enQueue(fakeMail())));
+    }
+
+    default void deQueueMail(Integer times) throws Exception {
+        MailQueue mailQueue = getMailQueue();
+
+        IntStream
+            .rangeClosed(1, times)
+            .forEach(Throwing.intConsumer(time -> mailQueue.deQueue().done(true)));
+    }
+
+    @Test
+    default void enqueueShouldIncreaseMetric() throws Exception {
+        Metric enqueuedMailsMetric = enqueuedMailsMetric();
+
+        enQueueMail(2);
+
+        verify(enqueuedMailsMetric, times(2)).increment();
```

Comment:
how do you know enqueuedMailsMetric is a mock ?

Reply:
At that contract level, No I don't, I created the mock at the impl level. Is this only the naming problem or some things else?

Reply:
i changed the variable name, if there is any other problem, please comment out :)

Reply:
please review file changes, this commit is old!